src/ore-instrument-macro/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! instrument macro with improved default safety
+//!
+//! This wraps the `tracing::instrument` macro and:
+//! - adds `skip_all`
+//! - errors on `skip`
+//!
+//! Its purpose is to prevent accidentally including large function arguments in tracing spans. By
+//! enforcing the use of skip_all, users must use the `fields` argument of the `tracing::instrument`
+//! macro to manually select their desired fields.
+
+extern crate proc_macro;
+
+use proc_macro::{TokenStream, TokenTree};
+
+#[proc_macro_attribute]
+pub fn instrument(attr: TokenStream, item: TokenStream) -> TokenStream {
+    // syn appears to not be able to parse the `%` part of things like `#[instrument(fields(shard =
+    // %id))]`, so we use the more naive proc_macro crate and look for strings.
```

Comment:
Interesting! Would be more than happy to pair here because using syn would be nice and allow us to provide compile errors with spans to point out exactly where the error is.

Reply:
Yes, let's pair. I probably did something wrong.

Reply:
The `tracing_attributes::instrument` macro uses their own syn parsing struct: https://github.com/tokio-rs/tracing/blob/master/tracing-attributes/src/attr.rs, not something from the syn crate. Wonder if this is what's going on.